dev-clean/headers/llsc_external.hpp

```diff
@@ -50,6 +56,36 @@ inline immer::flex_vector<std::pair<SS, PtrVal>> realloc(SS state, immer::flex_v
   return immer::flex_vector<std::pair<SS, PtrVal>>{{res, memLoc}};
 }
 
+inline std::string get_string(PtrVal ptr, SS state) {
+  std::string name;
+  char c = static_cast<char>(proj_IntV(state.at(ptr))); // c = *ptr
```

Comment:
truncation happens here -- probably add an assertion that checks the bit-width of `state.at(ptr)`.